File: AZ-204_Implement_Caching_For_Solutions/code/resource_mgmnt.cs
using Azure.Identity;
using Azure.ResourceManager.Redis;
using Azure.ResourceManager.Redis.Models;
using Azure.Core;

class Program
{
    static async Task Main(string[] args)
    {
        string subscriptionId = "<your-subscription-id>";
        string resourceGroupName = "<your-resource-group-name>";
        string cacheName = "<your-cache-name>";

        // Authenticate using Azure.Identity
        TokenCredential credential = new DefaultAzureCredential();

        // Create RedisManagementClient
        RedisManagementClient redisClient = new RedisManagementClient(subscriptionId, credential);

        // Provision Redis Cache
        RedisCreateParameters createParameters = new RedisCreateParameters
        {
            Location = "West US", // Specify the desired location
            Sku = new Sku("Standard", "C1"), // Specify the desired SKU
            EnableNonSslPort = false, // Set to true if you want to enable non-SSL port
            RedisConfiguration = new RedisConfiguration(new Dictionary<string, string>
            {
                { "maxmemory-policy", "allkeys-lru" } // Set any desired Redis configuration settings
            })
        };

        await redisClient.Redis.BeginCreateOrUpdateAsync(resourceGroupName, cacheName, createParameters);

        // Update Redis Cache
        RedisUpdateParameters updateParameters = new RedisUpdateParameters
        {
            Sku = new Sku("Standard", "C2"), // Update the SKU
            RedisConfiguration = new RedisConfiguration(new Dictionary<string, string>
            {
                { "maxmemory-policy", "volatile-lru" } // Update any desired Redis configuration settings
            })
        };

        await redisClient.Redis.BeginUpdateAsync(resourceGroupName, cacheName, updateParameters);

        // Delete Redis Cache
        await redisClient.Redis.BeginDeleteAsync(resourceGroupName, cacheName);
    }
}

File: AZ-204_Implement_Caching_For_Solutions/code/resource_mgmnt.py
from azure.identity import DefaultAzureCredential
from azure.mgmt.redis import RedisManagementClient
from azure.mgmt.redis.models import RedisCreateParameters, RedisUpdateParameters

# Set your Azure subscription ID and resource group name
subscription_id = 'your-subscription-id'
resource_group_name = 'your-resource-group-name'

# Set your Azure credentials
credential = DefaultAzureCredential()

# Set the Redis Cache name and location
redis_cache_name = 'your-redis-cache-name'
location = 'your-location'

# Create a Redis Cache
redis_client = RedisManagementClient(credential, subscription_id)
redis_params = RedisCreateParameters(location=location, sku={'name': 'Standard', 'family': 'C', 'capacity': 1})
redis_client.redis.create(resource_group_name, redis_cache_name, redis_params)

# Update the Redis Cache
redis_update_params = RedisUpdateParameters(sku={'name': 'Standard', 'family': 'C', 'capacity': 2})
redis_client.redis.update(resource_group_name, redis_cache_name, redis_update_params)

# Delete the Redis Cache
redis_client.redis.delete(resource_group_name, redis_cache_name)